sdks/wasm/Mono.WebAssembly.DebuggerProxy/DebugStore.cs

```diff
@@ -473,27 +474,54 @@ internal SourceFile (AssemblyInfo assembly, int id, Document doc, Uri sourceLink
 			} else {
 				this.Url = DotNetUrl;
 			}
-
 		}
 
 		internal void AddMethod (MethodInfo mi)
 		{
 			this.methods.Add (mi);
 		}
+
 		public string DebuggerFileName { get; }
 		public string Url { get; }
 		public string AssemblyName => assembly.Name;
-		public string DotNetUrl => $"dotnet://{assembly.Name}/{DebuggerFileName}";
-		public string DocHashCode => "abcdee" + id;
+		public string DotNetUrl => $"dotnet://{assembly.Name}{DebuggerFileName}";
+
 		public SourceId SourceId => new SourceId (assembly.Id, this.id);
 		public Uri SourceLinkUri { get; }
 		public Uri SourceUri { get; }
 
 		public IEnumerable<MethodInfo> Methods => this.methods;
+		public byte[] EmbeddedSource => doc.EmbeddedSource;
+
+		public (int startLine, int startColumn, int endLine, int endColumn) GetExtents ()
```

Comment:
I don't see this being used anywhere, so I'm guessing that this is for future work. Should the results be cached here, and invalidated if new methods are added?